src/main/java/com/faforever/client/remote/FafServerAccessorImpl.java

```diff
@@ -427,8 +433,22 @@ private void onSessionInitiated(SessionMessage sessionMessage) {
   }
 
   private void logIn(String username, String password) {
-    String uniqueId = uidService.generate(String.valueOf(sessionId.get()), preferencesService.getFafDataDirectory().resolve("uid.log"));
-    writeToServer(new LoginClientMessage(username, Hashing.sha256().hashString(password, UTF_8).toString(), sessionId.get(), uniqueId, localIp));
+    try {
+      String uniqueId = uidService.generate(String.valueOf(sessionId.get()), preferencesService.getFafDataDirectory().resolve("uid.log"));
+      writeToServer(new LoginClientMessage(username, Hashing.sha256().hashString(password, UTF_8).toString(), sessionId.get(), uniqueId, localIp));
+    } catch (IOException e) {
+      onUIDNotExecuted(e);
+    }
+  }
+
+  @VisibleForTesting
+  protected void onUIDNotExecuted(Exception e) {
+    if (e.getMessage() == null) {
+      return;
+    }
+    notificationService.addNotification(new ImmediateNotification(i18n.get("UIDNotExecuted"), e.getMessage(), Severity.ERROR,
+        Collections.singletonList(new ReportAction(i18n, reportingService, e))));
+    logger.error("UID.exe not executed", e);
```

Comment:
In case there is an exception without a message, it never gets logged and we'll never know anything about it (because `if(e.getMessage() == null) return;` above). If you move this log statement to the first line of the method you can solve this problem